apps/client/ui/Slider/Slider.tsx

```diff
@@ -1,82 +1,86 @@
+  {
+    backgroundColor: 'var(--primary-color)',
+    height: '14px',
+    width: '14px',
+    border: 'none',
+    opacity: 1,
+    boxShadow: '0px 4px 4px var(--shadow-grey)',
+  },
+  {
+    backgroundColor: 'var(--primary-color)',
+    height: '14px',
+    width: '14px',
+    border: 'none',
+    opacity: 1,
+    boxShadow: '0px 4px 4px var(--shadow-grey)',
+  },
+];
+
 export const CustomSlider: FC<CustomSliderProps> = ({
+  range = false,
   min = 0,
-  max = 500,
-  step = 50,
-  defaultValue = 250,
+  max = range ? 10 : 500,
+  step = range ? 1 : 50,
+  defaultValue = range ? [0, 10] : 250,
   className,
   ...props
 }) => {
-  const [value, setValue] = useState<number>(defaultValue);
-  const containerRef = useRef<HTMLDivElement>(null);
-
-  const handleOnChange = (newValue: number) => {
-    setValue(newValue);
-  };
-
-  const handleTooltipPosition = () => {
-    if (!containerRef.current) return;
-
-    const handle = containerRef.current?.getElementsByClassName('rc-slider-handle')[0] as HTMLElement | null;
-    const tooltip = containerRef.current?.getElementsByClassName('slider-tooltip')[0] as HTMLElement | null;
-    
-    if (handle && tooltip) {
-      const containerRect = containerRef.current.getBoundingClientRect();
-      const handleRect = handle.getBoundingClientRect();
-
-      const offset = handleRect.left - containerRect.left - handleRect.width;
-      tooltip.style.left = `${offset}px`;
-    }
+  const commonSliderProps = {
+    range,
+    min,
+    max,
+    step,
+    defaultValue,
+    className: 'w-full p-10',
+    handleRender: (origin, propsRender) => {
+      return (
+        <div className='relative'>
+          <div
+            className={clsx('w-[45px] py-1.5 px-2 mt-[-40px]', 'absolute -translate-x-1/2 z-[1] pointer-events-none', 'bg-dark-grey text-xs text-text-color text-center', 'border-0 rounded shadow shadow-shadow-grey')}
+            style={{
+              left: `${Math.round((propsRender.value / max) * 100)}%`,
+            }}
+          >
+            <span className={clsx('w-0 h-0 absolute bottom-[-4px] left-[50%] ml-[-4px]', 'border-transparent border-solid border-t-dark-grey border-t-4 border-x-4 border-b-0')}></span>
+            {propsRender.value}
+          </div>
+          {origin}
+        </div>
+      );
+    },
   };
 
-  useEffect(() => {
-    handleTooltipPosition();
-    window.addEventListener('resize', handleTooltipPosition);
-    return () => {
-      window.removeEventListener('resize', handleTooltipPosition);
-    };
-  }, [value]);
-
   return (
-    <div className='min-w-[180px] mt-12 mx-6 mb-0 h-9' ref={containerRef} {...props}>
+    <div className='min-w-[180px] mt-12 mx-6 mb-0 h-9' {...props}>
       <Slider
-        min={min}
-        max={max}
-        step={step}
-        defaultValue={value}
-        onChange={handleOnChange}
-        trackStyle={{ backgroundColor: 'var(--primary-color)', height: '4px' }}
+        {...commonSliderProps}
+        range={range}
```

Comment:
`range` here is redundant 👮 it is also in the `commonSliderProps` 😄 

...but if you will change styling of Slider's elements you would use only `commonSliderProps` or just simply pass props directly here instead of creating variable 👌 second option sounds better for me 😄